library/src/interactor_impl.cxx

```diff
@@ -78,158 +78,148 @@ class interactor_impl::internals
+        render = true;
         break;
       case 'F':
-        if (!renWin->GetFullScreen())
-        {
-          // save current window position and size
-          int* pos = renWin->GetPosition();
-          self->WindowPos[0] = pos[0];
-          self->WindowPos[1] = pos[1];
-          int* size = renWin->GetSize();
-          self->WindowSize[0] = size[0];
-          self->WindowSize[1] = size[1];
-        }
-
-        renWin->SetFullScreen(!renWin->GetFullScreen());
-
-        if (!renWin->GetFullScreen() && self->WindowSize[0] > 0)
-        {
-          // restore previous window position and size
-          renWin->SetPosition(self->WindowPos);
-          renWin->SetSize(self->WindowSize);
-        }
+        self->Options.toggle("fullscreen");
+        render = true;
+        /*        if (!renWin->GetFullScreen())
+                {
+                  // save current window position and size
+                  int* pos = renWin->GetPosition();
+                  self->WindowPos[0] = pos[0];
+                  self->WindowPos[1] = pos[1];
+                  int* size = renWin->GetSize();
+                  self->WindowSize[0] = size[0];
+                  self->WindowSize[1] = size[1];
+                }
+
+                renWin->SetFullScreen(!renWin->GetFullScreen());
+
+                if (!renWin->GetFullScreen() && self->WindowSize[0] > 0)
+                {
+                  // restore previous window position and size
+                  renWin->SetPosition(self->WindowPos);
+                  renWin->SetSize(self->WindowSize);
+                }
+
+                // when going full screen, the OpenGL context changes, we need to reinitialize
+                // the interactor, the render passes and the grid actor.
+                ren->ShowGrid(ren->IsGridVisible());
+                ren->SetupRenderPasses();
+                rwi->ReInitialize();
+
+                renWin->Render();*/ // TODO currently this breaks.
+        // should we just remove this hotkey as it is broken and nobody cares much about it ?
 
-        // when going full screen, the OpenGL context changes, we need to reinitialize
-        // the interactor, the render passes and the grid actor.
-        ren->ShowGrid(ren->IsGridVisible());
-        ren->SetupRenderPasses();
-        rwi->ReInitialize();
-
-        renWin->Render();
         break;
       case 'U':
-        ren->SetUseBlurBackground(!ren->UsingBlurBackground());
-        renWin->Render();
+        self->Options.toggle("blur-background");
+        render = true;
         break;
       case 'K':
-        ren->SetUseTrackball(!ren->UsingTrackball());
-        renWin->Render();
+        self->Options.toggle("trackball");
+        render = true;
         break;
       case 'H':
-        ren->ShowCheatSheet(!ren->IsCheatSheetVisible());
-        renWin->Render();
+        // Is this actually necessary ?
+        self->Options.toggle("cheatsheet");
+        render = true;
```

Comment:
do you need it @Meakk ?

Reply:
What do you mean? This feature is useful, don't you think?

Reply:
Yes, and I'm adding it, just checking this is fine with you :)

Reply:
It's fine 👍